mask/eolearn/mask/cloud_mask.py

```diff
@@ -292,3 +297,592 @@ def get_s2_pixel_cloud_detector(threshold=0.4, average_over=4, dilation_size=2,
+        temp_interweaved[..., 0::2] = temp_min
+        temp_interweaved[..., 1::2] = temp_mean
+
+        diff_interweaved = np.empty((*diff_max.shape[:-1], 2*diff_max.shape[-1]))
+        diff_interweaved[..., 0::2] = diff_max
+        diff_interweaved[..., 1::2] = diff_mean
+
+        # Put it all together
+        multi_features = np.concatenate((bands_t[nt_rel][None, ...],
+                                         loc_mu[nt_rel][None, ...],
+                                         ssim_interweaved,
+                                         temp_interweaved,
+                                         diff_interweaved
+                                         ),
+                                        axis=3
+                                        )
+
+        multi_features = multi_features.reshape(np.prod(multi_features.shape[:-1]), multi_features.shape[-1])
+
+        return multi_features
+
+    def execute(self, eopatch):
+        """
+        Add selected features (cloud probabilities and masks) to an EOPatch instance.
+
+        :param eopatch: Input `EOPatch` instance
+        :return: `EOPatch` with additional features
+        """
+
+        # Get data and is_data
+        feature_type, feature_name = next(self.data_feature(eopatch))
+        bands = eopatch[feature_type][feature_name][..., self.band_indices].astype(np.float32)
+
+        feature_type, feature_name = next(self.is_data_feature(eopatch))
+        is_data = eopatch[feature_type][feature_name].astype(bool)
+
+        original_shape = bands.shape[1:-1]
+        scale_factors, self.sigma = self._parse_resolution_data(original_shape, eopatch.meta_info)
+
+        mono_proba_feature, mono_mask_feature = self.mono_features
+        multi_proba_feature, multi_mask_feature = self.multi_features
+
+        # Downscale if specified
+        if scale_factors is not None:
+            bands = resize_images(bands.astype(np.float32), scale_factors=scale_factors)
+            is_data_sm = resize_images(is_data.astype(np.uint8), scale_factors=scale_factors).astype(np.bool)
+
+
+        mono_proba = None
+        multi_proba = None
+
+        # Run s2cloudless if needed
+        if any(feature is not None for feature in [self.mask_feature, mono_mask_feature, mono_proba_feature]):
+            mono_proba = self._mono_iterations(bands)
+            mono_proba = mono_proba.reshape(*bands.shape[:-1], 1)
+
+            # Upscale if necessary
+            if scale_factors is not None:
+                mono_proba = resize_images(mono_proba, new_size=original_shape)
+
+            # Average over and threshold
+            mono_mask = self._average_all(mono_proba) >= self.mono_threshold
+
+        # Run SSIM-based multi-temporal classifier if needed
+        if any(feature is not None for feature in [self.mask_feature, multi_mask_feature, multi_proba_feature]):
+            multi_proba = self._multi_iterations(bands, is_data_sm)
```

Comment:
`is_data_sm` might be unreferenced because it's defined under an `if` condition in line 835.